src/core/db/repository/user.py

```diff
@@ -44,6 +44,19 @@ async def set_categories_to_user(self, telegram_id: int, categories_ids: list[in
         if user:
             await self.update(user.id, user)
 
+    async def add_category_to_user(self, user: User, category_id: int) -> None:
+        """Добавляет категорию для пользователя."""
+        await self.create(UsersCategories(user_id=user.id, category_id=category_id))
+
+    async def delete_category_from_user(self, user: User, category_id: int) -> None:
+        """Удаляет категорию у пользователя."""
+        users_categories_obj = await self._session.scalar(
+            select(UsersCategories)
+            .where(UsersCategories.user_id == user.id)
+            .where(UsersCategories.category_id == category_id)
+        )
+        await self.remove(users_categories_obj)

```

Comment:
Можно сразу удалить, не делая запрос в бд. 
```suggestion
        users_categories_obj = await self._session.scalar(
            delete(UsersCategories)
            .where(UsersCategories.user_id == user.id)
            .where(UsersCategories.category_id == category_id)
        )
```
Не забудь `auto_commit`

Reply:
Поправил, заменил scalar на execute